### PR TITLE
Convert to Meson build system

### PR DESCRIPTION
**Pull request by Tomin1**
Convert sailjail to use Meson as its build system. Remove old Makefiles.
This should support roughly the same feature set that old make commands
did.

Move D-Bus policy file under /usr/share instead of /etc.

Also clean up any unused code and remove sailjail-plugin-devel
subpackage. Clean up spec file. Update outdated
firejail dependency version (templating support is required already).

Update out of date daemon/README as most missing stuff has
been already implemented. Also rename it to have an .md extension.

### CODE REVIEW

File: daemon/Makefile [hunk 1]
```diff
@@ -1,233 +0,0 @@
...
-# - - - - - - - - - - - - - - - - - - - -
-
-sailjail : sailjailclient
-	cp $< $@
-
-# - - - - - - - - - - - - - - - - - - - -
-# desktop_data
-# - - - - - - - - - - - - - - - - - - - -
-
-desktop_data : desktop_data.o util.o logging.o stringset.o
```
Comment (Tomin1) [inline daemon/Makefile:113]: I forgot this little tool from meson.build.

Comment (Tomin1) [reply daemon/Makefile:113]: Added it there. It's built with `meson compile -C build desktop_data` but not by default.

File: rpm/sailjail.spec [hunk 2]
```diff
@@ -75,71 +61,32 @@ Summary: Tests files for %{name}-daemon
...
+    -Dversion=%{version} \
+    -Duserunitdir=%{_userunitdir} \
+    -Dunitdir=%{_unitdir} \
+    -Ddbuspolicydir=%{_dbuspolicydir} \
+    -Dtestsdir=%{_testsdir} \
+
+%meson_build
 
 %install
 rm -rf %{buildroot}
```
Comment (dseight) [inline rpm/sailjail.spec:74]: I would propose to remove this line as well, as it's useless with modern rpm.

**Review by dseight — CHANGES_REQUESTED** [review]
Looks great! I have only several minor notes on this.

File: rpm/sailjail.spec (hunk 2 above)
Comment (Tomin1) [reply rpm/sailjail.spec]: Seems so. I removed it now.

File: BUILDING.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+It will print you URI to the generated report, assuming that you have
+the commands to build the reports in your system.
+
+Sailjail daemon diagram
+-----------------------
+There is an architecture diagram generated with graphviz and a png
+version of it can be generated with `sailjaild.png` target. You can find
+it in _build/daemon/_ after building it:
+
+    meson compile sailjaild.png -C build
```
Comment (llewelld) [inline BUILDING.md:40]: This one caused me quite a bit of trouble as we don't have graphviz available for `sb2`I couldn't find a solution without editing the build file to make it build outside of `sb2`.

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+Running unit tests
+------------------
+Unit tests can be run with meson:
+
+    meson test -C build
+
+After that you can also generate a coverage report if you built with
+coverage reports support:
+
+    ninja coverage -C build
```
Comment (llewelld) [inline BUILDING.md:29]: This also caused me trouble due to a missing `gconv` (I eventually got it to pip install in `sb2` and then everything worked fine, but is there a better way?).

**Review by llewelld — APPROVED** [review]
A few small comments (queries really). But this LGTM (I'm far from a meson expert though).

File: daemon/meson.build [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+migrator       = files('migrator.c')
+permissions    = files('permissions.c')
+prompter       = files('prompter.c')
+service        = files('service.c')
+session        = files('session.c')
+settings       = files('settings.c')
+stringset      = files('stringset.c')
+users          = files('users.c')
+util           = files('util.c')
+daemon_headers = include_directories('.')
```
[the remark this answers is not in the document]
Comment (Tomin1) [reply daemon/meson.build:129]: I would think that we would, at some point, want to extend the unit tests to cover more so I just thought to define them already. I think either approach is sound: we can only put here those that we need now, or just put everything like I've done.

File: BUILDING.md (hunk 3 above)
Comment (Tomin1) [reply BUILDING.md:40]: I built it outside our SDK. Works great with `-Dlibdbusaccess=disabled`.

File: BUILDING.md (hunk 4 above)
Comment (Tomin1) [reply BUILDING.md:29]: I built and run all unit tests outside our SDK. I'll just leave it for QA devices to run the QA versions of the tests on our platform.

File: daemon/meson.build (hunk 5 above)
[the remark this answers is not in the document]
Comment (llewelld) [reply daemon/meson.build:129]: Okay, it's fine to leave this as it is as far as I'm concerned then. Thanks.

File: BUILDING.md (hunk 4 above)
Comment (llewelld) [reply BUILDING.md:29]: Understood. I had trouble building outside the sdk due to missing dependencies, but I didn't try the `-D libdbusaccess=disabled` flag; thanks for explaining that to me.

File: BUILDING.md (hunk 3 above)
Comment (llewelld) [reply BUILDING.md:40]: That's the solution; thank you.

**Review by dseight — APPROVED** [review]

**Pull request merged**